Bugfix: Fix/forward type config

rossigee opened the pull request:
Fixes #32

tananaev commented:
Looks like this has unrelated commit.

rossigee commented:
Oops. Fixed.

tananaev commented on charts/traccar/values.yaml:
```diff
@@ -921,4 +921,4 @@ redis-ha:
     save: '""'
   haproxy:
-    enabled: true
+    enabled: true
```
Can we add the new line back here?

rossigee committed:
Fix forward configuration schema compatibility

- Update forward configuration to use proper type field
- Improve configuration schema structure for better compatibility
- Add configurable probe settings for deployment flexibility

rossigee commented:
@filippolmt - is this one of interest?

filippolmt commented on charts/traccar/templates/configmap.yaml:
```diff
@@ -260,8 +260,8 @@ data:
-{{- if .Values.traccar.forward.json }}
-        <entry key='forward.json'>{{ .Values.traccar.forward.json }}</entry>
+{{- if .Values.traccar.forward.type }}
+        <entry key='forward.type'>{{ .Values.traccar.forward.type }}</entry>
```
```suggestion
{{- $type := .Values.traccar.forward.type | default "url" -}}
{{- $allowed := list "url" "json" "amqp" "kafka" "mqtt" "redis" "wialon" -}}
{{- if not (has $t $allowed) }}
{{- fail (printf "forward.type '%s' is invalid. Allowed values: %v" $t $allowed) -}}
{{- end }}
        <entry key='forward.type'>{{ $type }}</entry>
{{- end }}
```

filippolmt replied:
I would probably do it like this.  
P.S. I haven’t tested it, these are just suggestions.

Anyway, even if I modified it, I don’t have the power to approve and merge your PR.

rossigee replied:
updated

rossigee committed:
Add validation for forward.type configuration values

rossigee committed:
Bump version.

rossigee commented:
The failing check is due to a missing/out-of-date image tag. Not related to the changes.

rossigee commented:
The test failure is nothing to do with this patch.

rossigee committed:
Merge branch 'main' into fix/forward-type-config
